sdk/azidentity/default_token_credential.go

```diff
@@ -21,6 +25,7 @@ type DefaultTokenCredentialOptions struct {
 // - ManagedIdentityCredential
 // Consult the documentation of these credential types for more information on how they attempt authentication.
 func NewDefaultTokenCredential(options *DefaultTokenCredentialOptions) (*ChainedTokenCredential, error) {
+	log := azcore.Log()
```

Comment:
The underlying credential types will log success/errors.  Do we think adding the same logging here is valuable?

Reply:
Well the logging I'm adding here is for the error that occurs when we cannot instantiate any credentials. 
I also thought it would be helpful to have the call for when DefaultTokenCredential invokes the NewChainedTokenCredential in the logs, because it might not be clear to some users why they are getting ChainedTokenCredential errors when they called DefaultTokenCredential. However, I will say that this information is in the comment of the function call, so I can see why this could be considered redundant. But I'm thinking through the scenario where the user is completely new to the identity library and does not understand exactly how DefaultTokenCredential works.

Reply:
That's a good point.  It might be good to run through a failure scenario here to see what the logging spew looks like.

Reply:
You're right, part of the log message was redundant in the error case. Here's the spew without the redundant error message:
```
Azure Identity => ERROR in NewEnvironmentCredential() call: Environment Credential: Missing environment variable AZURE_TENANT_ID
Azure Identity => ERROR in GetToken() call for NewDefaultTokenCredential(): an error was returned from an underlying credential type
```